### ISSUE TITLE
Feature Request: Add Email Verification for Registration and Password Recovery

### ISSUE THREAD

**Issue by anunol** [open-webui/open-webui#5717]
### Feature Request: Add Email Verification for Registration and Password Recovery

**Description:**
I would like to request the addition of an email verification feature to the Open WebUI project, which can be used for both user registration validation and password recovery. This feature would be enabled once the admin provides the necessary SMTP configuration. It should also allow admins to choose whether email verification is required for new user registrations, as well as include an option to enable a whitelist for specific email domains.

**Main Features:**

1. **Email Verification**:
   - When users register, they receive a verification email. They must validate their email address before their registration is completed.
   - When users forget their password, the system sends a reset link to their registered email.

2. **SMTP Configuration**:
   - Admins will be able to configure SMTP server settings, such as server address, port, email account, and password, via the admin panel.
   - The email verification feature will be activated only after successful SMTP configuration.

3. **Admin Option to Enable Email Verification for Registration**:
   - Admins can enable or disable email verification as a mandatory step for new user registrations.

4. **Whitelist for Email Domains**:
   - Admins can set a whitelist for email domains, allowing only specific domains to be used for registration.
   - This whitelist can be managed and updated by the admin through the settings.

**Use Case:**
- Enhances security by ensuring valid email addresses during user registration and recovery.
- Provides additional control for organizations that need to restrict registration to specific email domains.

**Expected Outcome:**
- Users will need to verify their email address to complete registration.
- Users can reset their password via email.
- Only emails from domains on the whitelist can be used to register new accounts.

**Relevant References:**
- Similar functions can refer to implementations in other projects (such as NextCloud's email verification system).

**Comment by pressdarling** [comment]
I don't think this project needs an email server running for all users. Maybe email verification hooks?

**Comment by justinh-rahb** [comment]
There was a previous effort in this direction that I attempted to steer towards a general solution for email hooks that could optionally be used in different parts of the project. Unfortunately it withered on the vine, but we'd still accept a PR that adds a facility for this functionality to be added.

Related:
- #2003

**Comment by pleabargain** [comment]
A way to recover passwords and regain access is very important. I have spent HOURS trying to get a stable instance of open-webui with a variety of emails and passwords but none are sticking.  Open-webui insists that I create a new user on each login as the last email/pwd are not accepted. 

How can I totally uninstall open-webui and all of its components that docker desktop generated? (Yes, I've googled the same question with no straight forward solution).

**Comment by pressdarling** [comment]
> How can I totally uninstall open-webui and all of its components that docker desktop generated? (Yes, I've googled the same question with no straight forward solution).

This is pretty straightforward. Safest way to do this is to use a different volume name for your open-webui installation - if you're using `docker run`, change `-v open-webui:/app/backend/data` to `-v some-other-name:/app/backend/data`. If that has the desired effect, then delete the original `open-webui` volume. 

Also, if this has happened twice, you should definitely be using a password manager! And note that without email functionality actually being baked in, you should be able to pick out any email address whether it exists or not (although I'd recommend one you can access just in case that changes).
